### DISCUSSION TITLE
Update temporal data with ValidFrom ValidTo

### DISCUSSION THREAD

**Discussion by laurent512** [prisma/prisma#14143] [Q&A]
Hello, 

Thank you so much for the great orm ! 

In my project, I havet to handle multiple Shops and allow them to update their properties. 
A constraint we have is to keep in DB the properties change made by those shops.

I reached the following model which seems quite convential for similar "temporal data"  (validTo = null means still valid today)

```
model Shop {
  id                    Int           @id @default(autoincrement())
  createdAt             DateTime      @default(now())
  updatedAt             DateTime      @updatedAt
  shopData              ShopData[]
}

model ShopData {
  id                 Int       @id @default(autoincrement())
  name               String
  bankAccount  String
  createdAt          DateTime  @default(now())
  updatedAt          DateTime  @updatedAt
  validFrom          DateTime  @default(now()) //inclusive
  validTo            DateTime? //exclusive
  shop               Shop      @relation(fields: [shopId], references: [id], onDelete: Cascade)
  shopId             Int
  // ... (more fields but simplified here)
}

```

To retrieve the currently valid ShopData I use the following : 
```
 select: {
              shopData: { where: { validTo: null }, select: { name: true } },
            },
```

To update the shopData, I have some difficulties as I would like to be a single query/transaction so that latency is reduced and both works or everything is cancelled.
Only solution I can find at the moment is not the one I want for the reasons exposed just above.

> Query 1 & 2 :  query to read latest shopData, query to update its validTo field 
> Query 3  :   query to create new shopData with copied information from Query 1&2 (putting validFrom at null, and changing the data requested by the user e.g. name)

Do you have a nice solution for this problem? :)

**Comment by pantharshit00** [comment]
Hello @laurent512 , our apologies for the delay in getting back to you! Due to the large number of questions in our community channels, unfortunately this slipped through the cracks.

Can you please share the three update queries you are performing here? Then I will be able to assess if we can rewrite those queries.

**Reply by vladi-stevanovic** [reply]
:wave: Hello @laurent512 , I’m going through pending GitHub Discussions to check if you need any help. Is this issue still open?

Please let us know if we can investigate it further or if you've already solved it, we'd love to know how!

It would be super useful for other users facing this problem! :raised_hands:

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
